src/lib/loaders.js

```diff
@@ -94,12 +95,39 @@ async function retrieveOrgInfo(orgName, token) {
     // eslint-disable-next-line id-length
     requiresTwoFactorAuthentication: organization.requiresTwoFactorAuthentication,
     twitterUsername: organization.twitterUsername,
-    websiteUrl: organization.websiteUrl
+    websiteUrl: organization.websiteUrl,
+    applications: await retrieveOrgApplications(orgName, token)
```

Comment:
IMHO don't do inline `awaits` like this in object construction.

Also, given #26, you might start to think about how the different rulesets will need to define which information they'll need to function, how to make sure you're only running each data-fetch once, and how the data gets provided to the different rules.

Reply:
Why not an inline `await`? Is that more of a preference/style thing, or for technical reasons?

As far as #26 though yeah I should probably make this line `applications: []`, and move this out to a separate later line of `result.applications = await ...`. Then later there can be something like `if (config.checkApplications) result.applications = await ...`.

Reply:
> Why not an inline await

There's not a technical reason, it works, but it can be hard to debug and hard to reason about imagine code like:

```js
const stuff = {
  foo: await foo(),
  bar: await bar(),
  fizzyLiftingDrinks: fizzyLiftingDrinks(),
  fizzbuzz: await fizzbuzz()
};
```

it can be hard to debug if there are issues. its hard to reason about the ordering of the promises and from looking at it it looks at first glance that it might be run in parallel, but that wouldn't be the case. Is the lack of `await` on the `fizzyLiftingDrinks` intentional or a bug?

Pulling the awaits out make it easier to reason about and the order more apparent and you could force it to be parallel if you wanted.